### PR TITLE
haskellPackages.yesod-core: fix build with Darwin sandbox enabled

### PR DESCRIPTION
**Pull request by reckenrode**
###### Description of changes

yesod-core needs to run a server on localhost during tests.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = relaxed` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/haskell-modules/configuration-darwin.nix [hunk 1]
```diff
@@ -117,6 +117,10 @@ self: super: ({
 
   yesod-bin = addBuildDepend darwin.apple_sdk.frameworks.Cocoa super.yesod-bin;
 
+  yesod-core = super.yesod-core.overrideAttrs (drv: {
+    __darwinAllowLocalNetworking = true;
```
Comment (cdepillabout) [inline pkgs/development/haskell-modules/configuration-darwin.nix:121]: Could you add a comment in the code here why this is needed?

Also, could someone that knows Darwin do a reveiw of this PR?  @NixOS/darwin-maintainers

Comment (reckenrode) [reply pkgs/development/haskell-modules/configuration-darwin.nix:121]: Sure. I’ll also do that for the http2 PR.

**Comment by mikroskeem** [comment]
Result of `nixpkgs-review pr 234129` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>datalad</li>
    <li>datalad.dist</li>
  </ul>
</details>
<details>
  <summary>8 packages failed to build:</summary>
  <ul>
    <li>git-annex (haskellPackages.git-annex)</li>
    <li>git-annex.doc (haskellPackages.git-annex.doc)</li>
    <li>haskellPackages.yesod-auth-hashdb</li>
    <li>haskellPackages.yesod-auth-hashdb.doc</li>
    <li>python310Packages.git-annex-adapter</li>
    <li>python310Packages.git-annex-adapter.dist</li>
    <li>python311Packages.git-annex-adapter</li>
    <li>python311Packages.git-annex-adapter.dist</li>
  </ul>
</details>
<details>
  <summary>90 packages built:</summary>
  <ul>
    <li>haskellPackages.bugsnag-yesod</li>
    <li>haskellPackages.bugsnag-yesod.doc</li>
    <li>haskellPackages.classy-prelude-yesod</li>
    <li>haskellPackages.classy-prelude-yesod.doc</li>
    <li>haskellPackages.espial</li>
    <li>haskellPackages.espial.doc</li>
    <li>haskellPackages.fluffy</li>
    <li>haskellPackages.fluffy.data</li>
    <li>haskellPackages.fluffy.doc</li>
    <li>haskellPackages.hledger-web</li>
    <li>haskellPackages.hledger-web.doc</li>
    <li>haskellPackages.hs-opentelemetry-instrumentation-yesod</li>
    <li>haskellPackages.hs-opentelemetry-instrumentation-yesod.doc</li>
    <li>haskellPackages.rowdy-yesod</li>
    <li>haskellPackages.rowdy-yesod.doc</li>
    <li>haskellPackages.serversession-frontend-yesod</li>
    <li>haskellPackages.serversession-frontend-yesod.doc</li>
    <li>haskellPackages.yesod</li>
    <li>haskellPackages.yesod-alerts</li>
    <li>haskellPackages.yesod-alerts.doc</li>
    <li>haskellPackages.yesod-auth</li>
    <li>haskellPackages.yesod-auth-basic</li>
    <li>haskellPackages.yesod-auth-basic.doc</li>
    <li>haskellPackages.yesod-auth-lti13</li>
    <li>haskellPackages.yesod-auth-lti13.doc</li>
    <li>haskellPackages.yesod-auth-oauth</li>
    <li>haskellPackages.yesod-auth-oauth.doc</li>
    <li>haskellPackages.yesod-auth-oauth2</li>
    <li>haskellPackages.yesod-auth-oauth2.doc</li>
    <li>haskellPackages.yesod-auth.doc</li>
    <li>haskellPackages.yesod-autoreload</li>
    <li>haskellPackages.yesod-autoreload.doc</li>
    <li>haskellPackages.yesod-core</li>
    <li>haskellPackages.yesod-core.doc</li>
    <li>haskellPackages.yesod-default</li>
    <li>haskellPackages.yesod-default.doc</li>
    <li>haskellPackages.yesod-elements</li>
    <li>haskellPackages.yesod-elements.doc</li>
    <li>haskellPackages.yesod-eventsource</li>
    <li>haskellPackages.yesod-eventsource.doc</li>
    <li>haskellPackages.yesod-fb</li>
    <li>haskellPackages.yesod-fb.doc</li>
    <li>haskellPackages.yesod-form</li>
    <li>haskellPackages.yesod-form-bootstrap4</li>
    <li>haskellPackages.yesod-form-bootstrap4.doc</li>
    <li>haskellPackages.yesod-form-bulma</li>
    <li>haskellPackages.yesod-form-bulma.doc</li>
    <li>haskellPackages.yesod-form-multi</li>
    <li>haskellPackages.yesod-form-multi.doc</li>
    <li>haskellPackages.yesod-form.doc</li>
    <li>haskellPackages.yesod-gitrepo</li>
    <li>haskellPackages.yesod-gitrepo.doc</li>
    <li>haskellPackages.yesod-json</li>
    <li>haskellPackages.yesod-json.doc</li>
    <li>haskellPackages.yesod-markdown</li>
    <li>haskellPackages.yesod-markdown.doc</li>
    <li>haskellPackages.yesod-media-simple</li>
    <li>haskellPackages.yesod-media-simple.doc</li>
    <li>haskellPackages.yesod-newsfeed</li>
    <li>haskellPackages.yesod-newsfeed.doc</li>
    <li>haskellPackages.yesod-page-cursor</li>
    <li>haskellPackages.yesod-page-cursor.doc</li>
    <li>haskellPackages.yesod-paginator</li>
    <li>haskellPackages.yesod-paginator.doc</li>
    <li>haskellPackages.yesod-persistent</li>
    <li>haskellPackages.yesod-persistent.doc</li>
    <li>haskellPackages.yesod-recaptcha2</li>
    <li>haskellPackages.yesod-recaptcha2.doc</li>
    <li>haskellPackages.yesod-routes-flow</li>
    <li>haskellPackages.yesod-routes-flow.doc</li>
    <li>haskellPackages.yesod-sitemap</li>
    <li>haskellPackages.yesod-sitemap.doc</li>
    <li>haskellPackages.yesod-static</li>
    <li>haskellPackages.yesod-static.doc</li>
    <li>haskellPackages.yesod-table</li>
    <li>haskellPackages.yesod-table.doc</li>
    <li>haskellPackages.yesod-tableview</li>
    <li>haskellPackages.yesod-tableview.doc</li>
    <li>haskellPackages.yesod-test</li>
    <li>haskellPackages.yesod-test.doc</li>
    <li>haskellPackages.yesod-websockets</li>
    <li>haskellPackages.yesod-websockets-extra</li>
    <li>haskellPackages.yesod-websockets-extra.doc</li>
    <li>haskellPackages.yesod-websockets.doc</li>
    <li>haskellPackages.yesod.doc</li>
    <li>haskellPackages.ytl</li>
    <li>haskellPackages.ytl.doc</li>
    <li>haskellPackages.yu-utils</li>
    <li>haskellPackages.yu-utils.doc</li>
    <li>hledger-web</li>
  </ul>
</details>

`nix run nixpkgs#nixpkgs-review -- pr 234129 --build-args '--relaxed-sandbox'`

**Comment by cdepillabout** [comment]
Thanks!

**Pull request merged**